README.md

```diff
@@ -1,2 +1,168 @@
+
+### /users/$userId$/devices/$deviceId$
+
+This data is required to send push notifications over the [Firebase Cloud Messaging](https://firebase.google.com/docs/cloud-messaging) System.
+
+|Property|Type|Values|Comments|
+|-|-|-|-|
+|modifiedDate|DateTime|-|This date is updated whenever the token is sent to the server, even if it is not replaced by a different token. It simply reflects the last date we can definitely confirm the token was active. Idea: We may ignore some devices, if the token has not been updated for a long time, since the app has not been opened for a long time.|
+|notificationToken|string|-|The FCM token as received from Firebase in the app. TBD: We may want to make this non-optional and simply not store device information, if there is no available token (e.g. if notifications are restricted).|
+|platform|optional string|e.g. "iOS", "Android"|This information is important as context for the `osVersion`, `appVersion` and `appBuild` properties.|
+|osVersion|optional string|e.g. "17.5.1"|The version of the OS. Depending on the OS, they may have different formats to be specified separately.|
+|appVersion|optional string|e.g. "1.0.1"|The version of the app as it is specified on the App/Play Store.|
+|appBuild|optional string|e.g. "56"|The build version of the app as defined by the CI pipeline releasing the app.|
+|language|optional string|e.g. "en"|Following IETF BCP-47 / [FHIR ValueSet languages](https://hl7.org/fhir/R4B/valueset-languages.html)|
+
+Q: Should we specify language/region by user or by device? Currently both a user and a device may contain a language. A notification shall prioritize a device's language setting, fall back to the user's language setting and use `en-US` in all other cases.
+
+#### Adding a new device or modifying information of an existing one
+
+A device can only be identified by its notification token. When updating a device's information, we therefore check if the notification token already exists in Firestore. If it already exists, we update all other existing fields' values - otherwise, we create a new device.
+
+A device may receive a different notification token at any time though. Therefore, we might create a new device, even though that device already exists in the table with a different (now inactive) notification token. Therefore, every time we send out a new notification and receive the information that a token is no longer active, we need to remove the device from this table.
+
+### /users/$userId$/messages/$messageId$
+
+|Property|Type|Values|Comments|
+|-|-|-|-|
+|dueDate|optional DateTime|-|The due date of the message to be shown in-app.|
+|triggerDate|optional DateTime|-|A date to trigger a notification on the device. If not set, then there is no notification to trigger. The notification should be triggered by the server. TBD: Are there cases where a message would trigger multiple times or event- rather than date-driven?|
```

Comment:
We would then add it multiple times.

I am wondering if we should generally skip this and generally trigger/add messages right when they should be sent out and when they would show up in the app. So e.g. for the daily reminder to measure your blood pressure we just have a scheduled cloud function that executes on a scheduled time (or once every hour if we consider time zones). We might need to therefore include the time zone in the device or even the user?

Reply:
I will add the question "When are messages shown to users and when are push notifications sent out to notify about them?" to the list.

Depending on the outcome, I'd be fine with simply storing a message for each push notifications sent out. I added a `type` property to easily identify a message type (e.g. a questionnaire reminder) to not bloat an inactive patient's messages when they are inactive for some time. Whenever we send out a new reminder, we can then simply remove all messages of that type and send out a new message/notification instead.

Reply:
I think part of that is answered in the UI document and development plan. I don't think we have a concrete time for them at this point so that's something that should be asked there.